usr/initiator.c

```diff
@@ -735,8 +735,13 @@ static void iscsi_login_eh(struct iscsi_conn *conn, struct queue_task *qtask,
 				session_conn_shutdown(conn, qtask, err);
 				break;
 			}
-			/* timeout during reopen connect. try again */
-			session_conn_reopen(conn, qtask, 0);
+			/*
+			 * stop connection for recovery if error is not
+			 * timeout
+			 */
+			if (err != ISCSI_ERR_PDU_TIMEOUT && err != ISCSI_ERR_TRANS_TIMEOUT)
+				stop_flag = STOP_CONN_RECOVER;
+			session_conn_reopen(conn, qtask, stop_flag);
```

Comment:
Nice debugging. We will not get a ISCSI_ERR_PDU_TIMEOUT error in this code path will we?

Reply:
> Nice debugging. We will not get a ISCSI_ERR_PDU_TIMEOUT error in this code path will we?

Yes, ISCSI_ERR_PDU_TIMEOUT would not happened if conn->state is ISCSI_CONN_STATE_XPT_WAIT.
My origin purpose is to handle all timeout sense. But it seems unnecessary, I would update

Reply:
> Nice debugging. We will not get a ISCSI_ERR_PDU_TIMEOUT error in this code path will we?

@mikechristie updated